codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/JsonShapeDeserVisitor.java

```diff
@@ -165,17 +167,22 @@ protected void deserializeStructure(GenerationContext context, StructureShape sh
                 String locationName = memberNameStrategy.apply(memberShape, memberName);
                 Shape target = context.getModel().expectShape(memberShape.getTarget());
 
+                String propertyAccess = getFrom("output", locationName);
+
                 if (usesExpect(target)) {
                     // Booleans and numbers will call expectBoolean/expectNumber which will handle
                     // null/undefined properly.
                     writer.write("$L: $L,",
                             memberName,
-                            target.accept(getMemberVisitor(memberShape, "output." + locationName)));
+                            target.accept(getMemberVisitor(memberShape, propertyAccess)));
                 } else {
-                    writer.write("$1L: (output.$2L !== undefined && output.$2L !== null)"
-                                    + " ? $3L: undefined,", memberName, locationName,
-                            // Dispatch to the output value provider for any additional handling.
-                            target.accept(getMemberVisitor(memberShape, "output." + locationName)));
+                    writer.write(
+                        "$1L: ($2L !== undefined && $2L !== null) ? $3L: undefined,",
+                        memberName,
+                        getFrom("output", locationName),
```

Comment:
Use `propertyAccess` here?
```suggestion
                        propertyAccess,
```